onnxruntime/core/providers/cuda/reduction/reduction_ops.cc

```diff
@@ -645,10 +645,9 @@ Status ReduceKernel<allow_multi_axes>::ComputeImpl(OpKernelContext* ctx, cudnnRe
   TensorShapeVector axes;
 
   size_t num_inputs = ctx->InputCount();
-  if (num_inputs == 2) {
+  const Tensor* axes_tensor = num_inputs == 2 ? ctx->Input<Tensor>(1) : nullptr;  // optional input. may be nullptr.
```

Comment:
I think we also need add code to handle `noop_with_empty_axes` for no axes like in ReduceSum-18.